packages/story-editor/src/app/story/effects/useHistoryEntry.js

```diff
@@ -103,6 +111,16 @@ function useHistoryEntry({ story, current, pages, selection, capabilities }) {
         skipAddingEntry =
           JSON.stringify(adjustedPages) === JSON.stringify(adjustedEntryPages);
       }
+
+      // skip entries that have a blob url
+      // https://github.com/GoogleForCreators/web-stories-wp/issues/10289
+      pages.map((page) => {
+        page.elements.forEach((element) => {
+          if (isBlobURL(element?.resource?.src)) {
+            skipAddingEntry = true;
+          }
+        });
+      });
```

Comment:
I'm not a big fan of this code residing here, but I guess it's okay for now.

I feel like the history provider shouldn't have all this domain-specific functionality (I know it had some before as well) and it would be better if it had a way of "registering" exception functions, and we'd then create the exceptions in the story provider where they belong. But I guess that's a bit too much work for this PR.